src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let joined = '';
+    let iterationCount = 0;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (const item of this) {
+      if (iterationCount < this.length - 1) {
+        if (item !== undefined && item !== null) {
+          joined += `${item}${separator}`;
+        } else {
+          joined += separator;
+        }
+      }
+      iterationCount++;
```

Comment:
You can avoid this by replacing `for ... of ...` loop with indexed loop.
```suggestion
      }
      
      iterationCount++;
```